src/core/aws-net-websocket-registry.adb

```diff
@@ -361,13 +386,16 @@ package body AWS.Net.WebSocket.Registry is
                exit when Read_Message (WebSocket, Message);
             end loop;
 
+            DB.Release_Socket (WebSocket);
+
          exception
             when E : others =>
                Do_Unregister (WebSocket);
                WebSocket_Exception
                   (WebSocket, Exception_Message (E), Protocol_Error);
                WebSocket.On_Close (Exception_Message (E));
                WebSocket.Shutdown;
```

Comment:
Shutdown and On_Close are already called by WebSocket_Exception and should not be called a second time I think

Reply:
Done here and some other places.